Riot/Modules/TabBar/TabBarCoordinator.swift

```diff
@@ -347,6 +349,10 @@ final class TabBarCoordinator: NSObject, TabBarCoordinatorType {
         }
         
         self.addMatrixSessionToMasterTabBarController(userSession.matrixSession)
+        
+        if let matrixSession = parameters.userSessionsService.mainUserSession?.matrixSession, matrixSession.groups().isEmpty {
```

Comment:
Maybe we can reuse the `showCommunitiesTab` property mentioned before. And a create an intermediate method: `func removeCommunitiesIfNeeded()` as we reuse the same mechanism in two different places.

Reply:
I cannot because it's slightly different for each case but I created the following var we can reuse
```
    private var currentMatrixSession: MXSession? {
        return parameters.userSessionsService.mainUserSession?.matrixSession
    }
```